src/dbwrapper.cpp

```diff
@@ -227,6 +227,7 @@ CDBWrapper::CDBWrapper(const DBParams& params)
     DBContext().iteroptions.fill_cache = false;
     DBContext().syncoptions.sync = true;
     DBContext().options = GetOptions(params.cache_bytes);
+    DBContext().options.max_file_size = params.options.max_file_size;
```

Comment:
setting the option here after we've partially constructed it seems inconsequential, could we add it as a parameter instead?
```diff
diff --git a/src/dbwrapper.cpp b/src/dbwrapper.cpp
--- a/src/dbwrapper.cpp	(revision 5f69dc11cb53c1794e9e71e6f398b95c1037cdc2)
+++ b/src/dbwrapper.cpp	(date 1727870700414)
@@ -134,9 +134,10 @@
              options->max_open_files, default_open_files);
 }
 
-static leveldb::Options GetOptions(size_t nCacheSize)
+static leveldb::Options GetOptions(size_t nCacheSize, size_t max_file_size)
 {
     leveldb::Options options;
+    options.max_file_size = max_file_size;
     options.block_cache = leveldb::NewLRUCache(nCacheSize / 2);
     options.write_buffer_size = nCacheSize / 4; // up to two write buffers may be held in memory simultaneously
     options.filter_policy = leveldb::NewBloomFilterPolicy(10);
@@ -226,8 +227,7 @@
     DBContext().iteroptions.verify_checksums = true;
     DBContext().iteroptions.fill_cache = false;
     DBContext().syncoptions.sync = true;
-    DBContext().options = GetOptions(params.cache_bytes);
-    DBContext().options.max_file_size = params.options.max_file_size;
+    DBContext().options = GetOptions(params.cache_bytes, params.options.max_file_size);
     DBContext().options.create_if_missing = true;
     if (params.memory_only) {
         DBContext().penv = leveldb::NewMemEnv(leveldb::Env::Default());
```

Reply:
That just seems uglier?